### PR TITLE
feat(cli): support for push command

### PR DESCRIPTION
**Pull request by htuna07**

### CODE REVIEW

**Comment by emre2345** [comment]
@htuna07 Do we have the same models with interface.ts in the api?

File: stacks/cli/src/interface.ts [hunk 1]
```diff
@@ -18,3 +18,54 @@ export abstract class Command extends BaseCommand<
...
+}
+
+export interface Dependency {
+  [key: string]: string;
+}
+
+export interface Asset {
+  kind: string;
+  metadata: MetaData;
+  spec: Function | any;
```
Comment (emre2345) [inline stacks/cli/src/interface.ts]: @htuna07 Doesn't this line mean spec is always "any"?

**Comment by htuna07** [comment]
> @htuna07 Do we have the same models with interface.ts in the api?
Actually yes but interface **Function** in api doesn't include **indexPath** and **dependencies** properties.

File: stacks/cli/src/interface.ts [hunk 2]
```diff
@@ -18,3 +18,54 @@ export abstract class Command extends BaseCommand<
...
+export interface MetaData {
+  name: string; //id
+}
+
+export interface LoginData {
+  token: string;
+  server: string;
+}
+
+export interface LoginData {
```
Comment (thesayyn) [inline stacks/cli/src/interface.ts]: There are two LoginData interfaces in this file. 👎

File: stacks/cli/src/interface.ts (hunk 1 above)
Comment (htuna07) [reply stacks/cli/src/interface.ts]: But I didn't know how to define this property before we decide asset types like **Bucket**, **Bucket Data** , **Function**..

Comment (htuna07) [reply stacks/cli/src/interface.ts]: But yes, it means always "any" like you said. What should we do then ?

Comment (emre2345) [reply stacks/cli/src/interface.ts]: @htuna07 We should use the types if we've them. You might use Any for now but create a tech story for us to discuss later.

**Comment by thesayyn** [comment]
@htuna07 Can you rebase and resolve conflicts?

**Pull request merged**